synapse/handlers/federation_event.py

```diff
@@ -1040,7 +1026,7 @@ async def _process_received_pdu(
         self,
         origin: str,
         event: EventBase,
-        state: Optional[Iterable[EventBase]],
+        state: Optional[StateMap[str]],
```

Comment:
As a general theme: I'd love it if we could change the generic `state` parameters to `state_ids_before_event` or similar, to make it easier to understand whether we are dealing with event ids or complete events, without having to go and look at the type.

Similarly let's rename methods which now return collections of event ids where they previously returned events.